src/storage/src/source/postgres/replication.rs

```diff
@@ -247,6 +248,8 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                         let update = ((oid, Err(err.clone())), MzOffset::from(u64::MAX), 1);
                         data_output.give(&data_cap_set[0], update).await;
                     }
+
+                    definite_error_handle.give(&definite_error_cap_set[0], ReplicationError::Definite(Rc::new(err))).await;
                     return Ok(());
```

Comment:
could we simplify the signature of `raw_stream` using `ReplicationError` as the error-return type and avoid the nested `Result`? Then we could unpack `ReplicationError` here to determine if it contains a `Transient` or `Definite` error, and act appropriately

Reply:
The outermost result lets us simply return to the closure if we encounter a transient error using `?`. Changing the signature of `raw_stream` would just ask of us to add another match arm that then used `?`/`return`––either approach is defensible because IMO they're essentially isomorphic. I prefer this to separate the transient and definite error channels.